### ISSUE TITLE
SDL 2.0.10+ causes massive slowdowns

### ISSUE THREAD

**Issue by ltriant** [ltriant/nes#6]
Frame rate drops from a constant 60 FPS to... something like 10 FPS?

The only way around this seems to be to downgrade to 2.0.9, which I've done like this:

```
$ curl -O https://libsdl.org/release/SDL2-2.0.9.tar.gz
$ tar zxvf SDL2-2.0.9.tar.gz
$ cd SDL2-2.0.9
$ ./configure --prefix=/usr/local/Cellar/sdl2/2.0.9 --without-x
$ make install
$ brew unlink sdl2
$ rm -rf /usr/local/Cellar/sdl2/2.0.10
$ brew link sdl2
```

Need to figure out why it's slowing down so much. Haven't looked into it at all.

**Comment by ltriant** [comment]
Calling `fill_rect` for each pixel is apparently slow. So I moved to updating a 2D vector representing the color of each of the 256x240 pixels. And then, upon hitting vblank (when we render to screen), I use that 2D vector to populate a texture and copy it onto the window.

This solved the frame-rate problem. But the time delay that incurred from locking the texture, copying it onto the canvas, and rendering causes all kinds of audio glitches, because that's time that audio isn't playing...

I tried drastically increasing the number of samples we keep in our buffer to offset this delay, but it doesn't work very well.

So I'm not happy with this for now.
